src/ApiException.php

```diff
@@ -138,9 +181,41 @@ public static function createFromRestApiResponse(
             $previous
         );
     }
+    
+    /**
+     * Checks if decoded metadata includes errorInfo message.
+     * @param array $decodedMetadata
+     * @return array $errorInfo {
+     *     @type boolean $containsErrorInfo
+     *     @type string $reason
+     *     @type string $domain
+     *     @type array $errorInfoMetadata
+     * }
+     */
+    public static function containsErrorInfo(array $decodedMetadata)
+    {
+        $errorInfo = [];
+        $errorInfo['containsErrorInfo'] = false;
+        if (empty($decodedMetadata)) {
+            return $errorInfo;
+        }
+        foreach ($decodedMetadata as $value) {
+            $isErrorInfoArray = array_key_exists('reason', $value);
```

Comment:
Not pretty sure what if other type has `reason` as key. Here is the code I used in Typescript, maybe it helps
```
const decodedDetail = this.decodeProtobufAny(detail);
          details.push(decodedDetail);
          if (detail.type_url === 'type.googleapis.com/google.rpc.ErrorInfo') {
            errorInfo = decodedDetail as unknown as ErrorInfo;
          }
          ```
https://github.com/googleapis/gax-nodejs/blob/2f3ac59564f112d501ceab5f3a319415e6cb80ef/src/googleError.ts#L211-L214

Reply:
Good point - I've updated it to check for `reason`, `domain`, and `metadata` since all three keys should exist if ErrorInfo exists.